app.py

```diff
@@ -0,0 +1,244 @@
+# Copyright 2024 D-Wave
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import argparse
+from typing import NamedTuple, Union
+
+import dash
+import diskcache
+from dash import MATCH, DiskcacheManager, ctx
+from dash.dependencies import Input, Output, State
+from dash.exceptions import PreventUpdate
+
+from app_configs import APP_TITLE, THEME_COLOR, THEME_COLOR_SECONDARY
+from dash_html import generate_problem_details_table_rows, set_html
+from src.enums import SamplerType
+
+cache = diskcache.Cache("./cache")
+background_callback_manager = DiskcacheManager(cache)
+
+# Fix Dash long callbacks crashing on macOS 10.13+ (also potentially not working
+# on other POSIX systems), caused by https://bugs.python.org/issue33725
+# (aka "beware of multithreaded process forking").
+#
+# Note: default start method has already been changed to "spawn" on darwin in
+# the `multiprocessing` library, but its fork, `multiprocess` still hasn't caught up.
+# (see docs: https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods)
+import multiprocess
+
+if multiprocess.get_start_method(allow_none=True) is None:
+    multiprocess.set_start_method("spawn")
+
+app = dash.Dash(
```

Comment:
What do you think about separating the code here into a file that the user shouldn't need to touch (could be `app.py`) containing e.g., the `app` specifics`, the CLI, CSS file update, and potentially even the toggle-left-column callback; and a second file with the callbacks which should be updated?